### DISCUSSION TITLE
lunr taking +4 seconds CPU time

### DISCUSSION THREAD

**Discussion by improvethings** [mmistakes/minimal-mistakes#2563] [Q&A] [answered]
## Environment

- Minimal Mistakes version: 4.19.3
- Ruby gem or remote theme version: 
- Jekyll version: 3.8.7
- Git repository URL:
- GitHub Pages hosted (if yes provide URL to site):
- Operating system: Debian 10

## Expected behavior

I expect the website, with lunr, to load in less than a second.

## Steps to reproduce the behavior

Chrome now has a website audit tool built in. Go to Inspect Element -> Audits/Lighthouse -> Generate Report. lunr is taking +4 seconds. 

## Other

I initially posted it to lunr, they suggest it's an mm configuration issue: 

https://github.com/olivernn/lunr.js/issues/465

$ ls -l _posts/ | wc -l
366

**Comment by mmistakes** [comment]
Link to your repo so I can inspect. 
Chances are it’s your index. If you have a lot of posts or indexing full content it can produce a large file that needs to be pulled down from the server impact page load performance.

Ref MM config docs:

> Note: Only the first 50 words of a post or page’s body content is added to the Lunr search index. Setting `search_full_content` to `true` in your _config.yml will override this and could impact page load performance.

**Comment by improvethings** [comment]
Sent you a DM on Twitter to the private repo. Thanks to the pointer to your docs, I do have search_full_content enabled, as only the first 50 words isn't a very effective search query when most articles are several hundred words long.

**Comment by mmistakes** [comment, marked as answer]
That’s your issue then. Using full builds a search index file that has the content of **every** post. That’s being loaded on every page. I bet if you inspect that file it’s huge and that’s the problem.

Either use excerpts or move to a more performant solution like Algolia. That’s what I use for the demo site. Way more configurable and better search results.
